io/private/copcwriter/Common.hpp

```diff
@@ -0,0 +1,135 @@
+/******************************************************************************
+* Copyright (c) 2021, Hobu Inc. (info@hobu.co)
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#pragma once
+
+#include <stdint.h>
+#include <array>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <pdal/PointTable.hpp>
+#include <pdal/Scaling.hpp>
+#include <pdal/SpatialReference.hpp>
+#include <pdal/util/Bounds.hpp>
+
+#include <filters/StatsFilter.hpp>
+#include <io/HeaderVal.hpp>
+
+#include <io/private/las/Utils.hpp>
+
+#include <lazperf/vlr.hpp>
+
+namespace pdal
+{
+
+namespace stats
+{
+
+enum class Index
+{
+    X = 0,
+    Y,
+    Z,
+    GpsTime,
+    ReturnNumber
+};
+
+} // namespace stats
+
+namespace copcwriter
+{
+
+// These are hopes, not absolutes.
+const int MaxPointsPerNode = 100000;
```

Comment:
What are thoughts on allowing people to manipulate these with options?

Reply:
If it's up to me I say wait until someone demonstrates a need.

Reply:
This should be configurable, it makes a large difference when considering loading over web with potree or such.

Reply:
We are not going to make this configurable for the upcoming release, but we may revisit it subsequently. People who need more flexibility can use copc-lib, etc to make files for now.